File: src/main/java/com/mbayou/kick4k/moderation/PostModerationBansRequest.java
package com.mbayou.kick4k.moderation;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class PostModerationBansRequest {
    private final Integer broadcasterUserId;
    private final Integer duration;
    private final String reason;
    private final Integer userId;

    @JsonCreator
    public PostModerationBansRequest(@JsonProperty("broadcaster_user_id") Integer broadcasterUserId,
                                     @JsonProperty("duration") Integer duration,
                                     @JsonProperty("reason") String reason,
                                     @JsonProperty("user_id") Integer userId) {
        this.broadcasterUserId = broadcasterUserId;
        this.duration = duration;
        this.reason = reason;
        this.userId = userId;
    }

    public Integer getBroadcasterUserId() {
        return broadcasterUserId;
    }

    public Integer getDuration() {
        return duration;
    }

    public String getReason() {
        return reason;
    }

    public Integer getUserId() {
        return userId;
    }

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private Integer broadcasterUserId;
        private Integer duration;
        private String reason;
        private Integer userId;

        public Builder broadcasterUserId(Integer broadcasterUserId) {
            this.broadcasterUserId = broadcasterUserId;
            return this;
        }

        public Builder duration(Integer duration) {
            this.duration = duration;
            return this;
        }

        public Builder reason(String reason) {
            this.reason = reason;
            return this;
        }

        public Builder userId(Integer userId) {
            this.userId = userId;
            return this;
        }

        public PostModerationBansRequest build() {
            if (broadcasterUserId == null) {
                throw new IllegalStateException("BroadcasterUserId is required");
            }

            if (userId == null) {
                throw new IllegalStateException("UserId is required");
            }

            return new PostModerationBansRequest(broadcasterUserId, duration, reason, userId);
        }
    }
}

File: src/main/kotlin/com/mbayou/kick4k/moderation/PostModerationBansRequest.kt
package com.mbayou.kick4k.moderation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_NULL)
data class PostModerationBansRequest @JsonCreator constructor(
    @JsonProperty("broadcaster_user_id") val broadcasterUserId: Int,
    @JsonProperty("duration") val duration: Int?,
    @JsonProperty("reason") val reason: String?,
    @JsonProperty("user_id") val userId: Int,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var broadcasterUserId: Int? = null
        private var duration: Int? = null
        private var reason: String? = null
        private var userId: Int? = null

        fun broadcasterUserId(broadcasterUserId: Int) = apply { this.broadcasterUserId = broadcasterUserId }
        fun duration(duration: Int?) = apply { this.duration = duration }
        fun reason(reason: String?) = apply { this.reason = reason }
        fun userId(userId: Int) = apply { this.userId = userId }

        fun build(): PostModerationBansRequest {
            val broadcasterId = requireNotNull(broadcasterUserId) { "BroadcasterUserId is required" }
            val userIdValue = requireNotNull(userId) { "UserId is required" }
            return PostModerationBansRequest(broadcasterId, duration, reason, userIdValue)
        }
    }
}
